cmd/kogito/command/deploy/deploy_runtime.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"fmt"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/common"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/context"
+	"github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/shared"
+	buildutil "github.com/kiegroup/kogito-cloud-operator/cmd/kogito/command/util"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type deployRuntimeFlags struct {
+	common.DeployFlags
+	common.InfinispanFlags
+	common.KafkaFlags
+	name              string
+	image             string
+	runtime           string
+	enableIstio       bool
+	enablePersistence bool
+	enableEvents      bool
+	serviceLabels     []string
+}
+
+type deployRuntimeCommand struct {
+	context.CommandContext
+	command *cobra.Command
+	flags   deployRuntimeFlags
+	Parent  *cobra.Command
+}
+
+// initDeployCommand is the constructor for the deploy command
+func initDeployRuntimeCommand(ctx *context.CommandContext, parent *cobra.Command) context.KogitoCommand {
+	cmd := &deployRuntimeCommand{CommandContext: *ctx, Parent: parent}
+	cmd.RegisterHook()
+	cmd.InitHook()
+	return cmd
+}
+
+func (i *deployRuntimeCommand) Command() *cobra.Command {
+	return i.command
+}
+
+func (i *deployRuntimeCommand) RegisterHook() {
+	i.command = &cobra.Command{
+		Use:     "deploy-runtime NAME [IMAGE]",
```

Comment:
If we are going to stick with this new command, I'd say to just use "deploy-service" instead. Recently we renamed everything from `runtime` to service in the docs/description. So this is a custom Kogito service. Not a "runtime" any more. It's ok to keep this name internally, but since this is part of the UX, I'd say to migrate everything to `deploy-service` with the `deploy` alias.

Reply:
currently deploy-service is already used to create kogito-app. If I used same cmd to deploy runtime in it will be conflicting

Reply:
This JIRA is to get rid of the `KogitoApp`, that's why I ask.

Reply:
The point is, the `deploy` should create a `KogitoRuntime` and a `KogitoBuild` CRs together. So we will have a build and deployment controlled by k8s once the CLI deploys both.

Reply:
Of course, in some cases you just create the `KogitoRuntime`, if an image is passed to the CLI, for example.

Reply:
@ricardozanini `KogitoRuntime` can independently run on K8 but `KogitoBuild` depends on openshift component(like ImageStream). 
For K8 user only `KogitoRuntime` is helpful, If we combine both of them in single cmd then it could be confusing for K8 user.
Could we keep them as 2 different cmd?

Reply:
Like I said above, we can have both commands in the `deploy` command:

Without build:

`kogito deploy <image> [OPTIONS]`

With build:

`kogito deploy <source> [OPTIONS]`

I don't think it's confusing. The cli should be able to fetch the discovery API and warn the user that deploying from source is not supported on k8s yet.

Reply:
I had merge both `kogito deploy-runtime` and `kogtio build_service` cmd into single `kogito deploy-service` cmd with alias `deploy`